### ISSUE TITLE
TwitchLib channel events consistency suggestion

### ISSUE THREAD

**Issue by Danatobob** [swiftyspiffy/TwitchLib#85]
I'm trying to build a collection of active channel users by using TwitchLib events. The idea is to have a dictionary with username as key and add/remove entry depending on events.
TwitchLib has the following events for this:

- OnExistingUsersDetected; ExistingUsers - Property representing string list of existing **users**
- OnViewerJoined; Username - Property representing **username** of joined viewer.
- OnViewerLeft; Username - Property representing **username** of user that left.
- OnViewerBanned; Viewer - **Viewer** that was banned.
- OnViewerTimedout; Viewer - **Viewer** that was timedout.

Now I wasn't really sure those all mean the same thing. Logically they should, but why are the property names different then? Maybe twitch sends different stuff (DisplayName or smth) and TwitchLib obeys ? Only after checking the [TwitchIRC specs](https://github.com/justintv/Twitch-API/blob/master/IRC.md) I've found out that all those commands use twitch_username parameters so those events give me the right data to use as the key value for my dictionary.

So, the suggestion is to either clarify in the docs of the property that Viewer/ExistingUsers are username(s), or perhaps even rename some of those properties.

**Comment by swiftyspiffy** [comment]
Please see changes made in this commit on branch 85 to confirm everything is changed as requested..

https://github.com/swiftyspiffy/TwitchLib/commit/aec13d946c143d8a841965fcb6a52cdbfce02ac6

**Comment by Danatobob** [comment]
Seems perfect now. Thank you.

**Comment by swiftyspiffy** [comment]
Awesome, merged, closing this now :D

**Comment by Danatobob** [comment]
Hi. Didn't want to create new issue for a minor thing related to this, but I've noticed some inconvenience with the order of OnChatCommandReceived and OnMessageReceived event firing. Since Twitch delays user joined/left notifications up to 2 minutes I usually end up receiving chat commands from users not yet registered in my list. This is important because some of the commands are vote-based with an overall percentage threshold so I must know the total amount of users that are in the chat before processing the command. 
The old solution was to add users to my list right in the OnChatCommandReceived event handler. Now I need to add them when they type any message in the chat, not just chat command, so I could detect their presence earlier. And that's where the issue is: OnMessageReceived fires after OnChatCommandReceived, so if I only use on-message event handler, the user would not yet exist in the list when on-chat-command is called. I could fix that by putting my "add user to list" code in both events, but it would mean it will always execute twice when user types a command. I think its counter-intuitive that on-message is not the first event that is fired because that's what is initially received and on-chat-command is an additional logic applied to the received message. Was there a reason for those events to fire in that order ?

**Comment by swiftyspiffy** [comment]
There is no reason for the current order. I can fix this now though.

**Comment by swiftyspiffy** [comment]
Committed: 2333c0c642a499aeb285a8fca7c887fc30407b62
